### ISSUE TITLE
Volume driver running as a container hits volume create timeouts during restart

### ISSUE THREAD

**Issue by srust** [docker/docker#21300]
**Output of `docker version`:**

```
Client:
 Version:      1.11.0-dev
 API version:  1.23
 Go version:   go1.6
 Git commit:   45b2a57-unsupported
 Built:        Thu Mar 17 15:50:07 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.0-dev
 API version:  1.23
 Go version:   go1.6
 Git commit:   45b2a57-unsupported
 Built:        Thu Mar 17 15:50:07 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
docker info
Containers: 7
 Running: 1
 Paused: 0
 Stopped: 6
Images: 184
Server Version: 1.11.0-dev
Storage Driver: devicemapper
 Pool Name: docker-253:1-36654375-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 107.4 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 43.05 GB
 Data Space Total: 107.4 GB
 Data Space Available: 6.089 GB
 Metadata Space Used: 24.45 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.123 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Steps to reproduce the issue:**
1. Start a volume driver running in a container, and create at least one volume (ie: activate the plugin)
2. NOTE: The volume driver container uses local volumes itself for its own state, etc.
3. Restart volume driver

**Describe the results you received:**

See timeouts contacting driver. Eventually after retry timeout (for each local volume the driver uses) it starts up.

**Describe the results you expected:**

Previously (docker <= 1.10), a restart of a driver worked immediately. For development or upgrade of a driver, this is an important operation.

**Additional information you deem important (e.g. issue happens only occasionally):**

Because the driver was started and registered, every new volume create checks to see if that driver has the volume. I believe this was introduced in #20198, where a GET is now performed to see if the driver has that volume. Since the driver / plugin is restarting it cannot respond to GET requests.

Questions:

A) would this be helped by the new plugin infrastructure (docker plugin) and related commands?

B) Can the driver be unregistered during restart, so it will not be queried. Clearly the volumes it is using are not owned by itself...

C) Other options?

Thanks.

**Comment by thaJeztah** [comment]
ping @cpuguy83 @calavera could you have a look at this? Is this something to be fixed / looked at before 1.11?

**Comment by srust** [comment]
Ok I'm going to close this. Managed volume plugins makes this moot. Legacy plugins will "deal" with it.
